kartothek/core/dataset.py

```diff
@@ -384,11 +384,9 @@ def get_indices_as_dataframe(
         columns_to_scan = columns[:]
         if predicates:
             predicate_columns = columns_in_predicates(predicates)
-            # Don't use set logic to preserve order
-            # TODO why do I need to preserve the order here?
-            for col in predicate_columns:
-                if col not in columns_to_scan and col in self.indices:
-                    columns_to_scan.append(col)
+            columns_to_scan = sorted(
+                predicate_columns.intersection(self.indices.keys()).union(columns)
```

Comment:
nitpick, I think this improves legibility, feel free to ignore
```suggestion
                (predicate_columns & self.indices.keys()) | set(columns)
```